alchemist/src/strategies/basic.py

```diff
@@ -0,0 +1,103 @@
+
+import pandas as pd
+
+from alchemist.src.etl.data_loader import load_market_data
+from alchemist.src.etl.technical_indicators import PriceBySMA, BBP
+from alchemist.src.consts import *
+
+from datetime import datetime
+import datetime as dt 
+
+from alchemist.src.strategies.trader import Trader
+from pandas.core.frame import DataFrame
+
+class HODLer(Trader):
+    def __init__(self, name: str, impact: float=0.0):
+        super().__init__(name, impact)
+    
+    def trade(self, symbol: str, start_date: datetime=datetime(2020, 1, 1), end_date: datetime=datetime(2020, 12, 31), start_value: float=10000) -> DataFrame:
+
+        # Loading daily market data
+        data = load_market_data([symbol], start_date, end_date, return_dict=False, invalidate_cache=False)[symbol]
+
+        trades = pd.DataFrame(0, index=data.index, columns=[symbol, 'USD'])
+
+        start_date = trades.index[0]
+
+        # Buy Once 
+        price = data.loc[start_date, ADJUSTED_CLOSE]
+        units = start_value / price
+
+        trades.loc[start_date, symbol] = units
+        trades.loc[start_date, 'USD'] = -1*start_value 
+
+        return trades
+
+class BasicTrader(Trader):
+
+    def __init__(self, name: str, impact: float=0.0):
+        super().__init__(name, impact)
+        
+    def trade(self, symbol: str, start_date: datetime=datetime(2020, 1, 1), end_date: datetime=datetime(2020, 12, 31), start_value: float=10000) -> DataFrame:
+        
+        # Loading daily market data
+        daily_market_data = load_market_data(
+        [symbol], start_date, end_date, return_dict=False, invalidate_cache=False)
+
+        # Create instance of Price/SMA calculator
+        price_by_sma = PriceBySMA(name='Price_by_SMA_15d', window_size=15, price_column_name=ADJUSTED_CLOSE)
+        daily_market_data = price_by_sma.calculate(daily_market_data, merge_with_market_data=True)
+
+        # Create instance of BBP calculator
+        bbp = BBP(name='bbp_15d', window_size=15, price_column_name=ADJUSTED_CLOSE)
+        data = bbp.calculate(daily_market_data, merge_with_market_data=True)[symbol]
+
+        trades = pd.DataFrame(0, index=data.index, columns=[symbol, 'USD'])
+
+        trading_days = trades.index
+        position_status = 'close'
+
+        current_usd = start_value 
+        current_units = 0
+        
+        for td in trading_days:
+
+            price = data.loc[td, ADJUSTED_CLOSE]
+
+            if position_status == 'close':
+
+                # BUY
+                if (data.loc[td, 'Price_by_SMA_15d'] < 0.95) | (data.loc[td, 'bbp_15d'] < 0.2):
+                    units = current_usd/price
```

Comment:
I believe that line should be:
units = current_usd // price
otherwise you would trade fractions of stocks.

Reply:
Yeah, I think fractional shares are okay with most brokerages. Also cryptos are fractional.